torch/csrc/distributed/rpc/PythonRpcHandler.cpp

```diff
@@ -0,0 +1,89 @@
+#include <torch/csrc/distributed/rpc/PythonRpcHandler.h>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+PyObject* PythonRpcHandler::module_ = nullptr;
+PyObject* PythonRpcHandler::runUDFFunction_ = nullptr;
+PyObject* PythonRpcHandler::loadResultFunction_ = nullptr;
+
+void PythonRpcHandler::init() {
+  Py_Initialize();
+  module_ = PyImport_ImportModule("torch.distributed.internal_rpc_utils");
+  if (module_ == nullptr) {
+    throw std::runtime_error("import internal_rpc_utils module is NULL");
+    PyErr_Print();
+  }
+  runUDFFunction_ = PyObject_GetAttrString(module_, "run_python_udf_internal");
+  if (runUDFFunction_ == nullptr) {
+    throw std::runtime_error("import run_python_udf_internal is NULL");
+    PyErr_Print();
+  }
+  loadResultFunction_ = PyObject_GetAttrString(
+    module_, "load_python_udf_result_internal");
+  if (loadResultFunction_ == nullptr) {
+      throw std::runtime_error(
+        "import load_python_udf_result_internal is NULL");
+      PyErr_Print();
+  }
+}
+
+void PythonRpcHandler::cleanUp() {
+  Py_DECREF(module_);
+  Py_DECREF(runUDFFunction_);
+  Py_DECREF(loadResultFunction_);
+  Py_Finalize();
```

Comment:
This "kill"s Python interpreter.

https://docs.python.org/3/c-api/init.html#c.Py_FinalizeEx

Reply:
I added a flag to ensure we will call Py_Finalize() only if Py_initialize() is called in embedder